source/slides/03_linux_basics.rst

```diff
@@ -0,0 +1,443 @@
+
+\/usr (Historical Context)
+--------------------------
+
+People were running out of disk space so:
+
+* All binaries not required for base system + booting + mounting other
+  devices went in ``/usr/bin`` and ``/usr/sbin``
+* These binaries were typically manually compiled and installed by the user
+* Eventually some unices (linux didn't exist yet) took over
+  ``/usr/bin`` and ``/usr/sbin`` for the location that packages were
+  installed to.
+* Now manually installed (without package manager) binaries go in
+  ``/usr/local/bin`` and ``/usr/local/sbin``.
+
+\/usr (Modern Context)
+----------------------
+
+.. code-block:: bash
+
+  $ ls /usr
+  bin  games    lib    libexec  sbin   src
+  etc  include  lib64  local    share  tmp
+
+.. csv-table::
+  :header: Location, Description
+
+  /usr/bin,Packages installed by package manager
+  /usr/sbin,Packages installed by package manager
+  /usr/etc,Rarely used; files from /etc can be symlinked here
+  /usr/games,Binaries for games and educational programs
+
+\/usr (Modern Context)
+----------------------
+
+.. csv-table::
+  :header: Location, Description
+
+  /usr/include,Include files for the C compiler
+  /usr/lib,Object libraries (including dynamic libs); some unusual binaries
+  /usr/lib64,64-bit libraries
+  /usr/libexec,Executables used with libraries; not used much
+  /usr/local,Programs (and their configuration) locally installed by user go here
+
+\/usr (Modern Context)
+----------------------
+
+.. csv-table::
+  :header: Location, Description
+
+  /usr/share,Application data; typically examples and documentation
+  /usr/src/linux,Kernel source goes here
+
+\/dev
+-----
+
+* Device files, which often refer to physical devices
+
+  * ``/dev/sd?``
+  * ``/dev/sr?``
+  * ``/dev/tty*``
+
+* Special character devices:
+
+  * ``/dev/null`` -- sink for writes
+  * ``/dev/random`` -- high quality randomness (blocking)
+  * ``/dev/urandom`` -- non-blocking random
+  * ``/dev/zero`` -- always reads 0s
+
+\/etc
+-----
+
+* Configuration files local to the machine
+* Programs almost always look here for configuration first
+
+\/home
+------
+
+* Contains homedirs of regular users
+* Sometimes symlinked to ``/usr/home``, but rarely on linux
+
+\/lib & \/lib64
+---------------
+
+* Libraries needed to boot and run commands related to bootstrapping
+
+\/media & \/mnt
+---------------
+
+* Used as mount points for other devices (usb sticks, nfs, etc)
+* Most Desktop Environments automatically mount things to ``/media``
+
+\/proc
+------
+
+* Special filesystem ``procfs`` contains a file-representation of
+  the current state of the kernel and running processes.
+
+\/sys
+-----
+
+* File-representation of device drivers, subsystems, and hardware
+  loaded into the kernel
+* Similar to ``sysctl`` on other Unixy systems
+
+\/var
+-----
+
+* Multi-purpose: log, temporary, transient, and spool files
+* Typically contains run-time data
+* Cache
+
+Devices
+-------
+
+* TODO
```

Comment:
I wasn't quite sure. I removed some duplicated content I went over last week. I was contemplating going into more detail about how devices work on linux and mention udev, etc. Did you have anything you'd like to cover?